packages/apsara-ui/src/Listing/InfiniteScroll.tsx

```diff
@@ -0,0 +1,91 @@
+import { debounce } from "lodash";
+import React, { useState, useEffect } from "react";
+
+interface InfiniteScrollProps<T> {
+    fetchMoreData: (page: number, pageSize: number, filters: any) => Promise<T[]>;
+    contentRef: React.RefObject<HTMLDivElement>;
+    page: number;
+    pageSize?: number;
+    filters?: any;
+    threshold?: number;
+    renderItem: (item: T) => React.ReactNode;
+    loadingComponent?: React.ReactNode;
+    noMoreDataComponent?: React.ReactNode;
+}
+
+const InfiniteScroll: React.FC<InfiniteScrollProps<any>> = ({
+    fetchMoreData,
+    contentRef,
+    page,
+    pageSize = 10,
+    threshold = 0,
+    filters,
+    renderItem,
+    loadingComponent,
+    noMoreDataComponent,
+}) => {
+    const [data, setData] = useState<any[]>([]);
+    const [isLoading, setIsLoading] = useState(false);
+    const [hasMoreData, setHasMoreData] = useState(true);
+
+    useEffect(() => {
+        setData([]);
+        setHasMoreData(true);
+    }, [filters]);
+
+    const fetchMore = debounce(async () => {
+        if (!isLoading && hasMoreData && isBottom(contentRef)) {
+            setIsLoading(true);
+            try {
+                const newData = await fetchMoreData(page, pageSize, filters);
+                setData((prevData) => prevData.concat(newData));
+                setHasMoreData(newData.length < pageSize ? false : true);
+            } finally {
+                setIsLoading(false);
+            }
+        }
+    }, 200);
+
+    useEffect(() => {
```

Comment:
this effect will subscribe and unsubscribe every render.

we will need to design the hooks per scenario and not clumped them together.

ideally we only want to add listener only when the container change.